concent_api/core/subtask_helpers.py

```diff
@@ -162,7 +163,7 @@ def check_compatibility(subtask: Subtask, client_public_key: bytes) -> None:
     if not is_protocol_version_compatible(subtask.task_to_compute.protocol_version):
         error_message = f'Unsupported version of golem messages in stored messages. ' \
             f'Version stored in database is {subtask.task_to_compute.protocol_version}, ' \
-            f'Concent version is {settings.GOLEM_MESSAGES_VERSION}.'
+            f'Concent version is {get_major_and_minor_golem_messages_version(settings.GOLEM_MESSAGES_VERSION)}.'
```

Comment:
Why not making this a constant? The version should not change without a server restart. If you make it a constant it will be calculated only once.

Reply:
@rwrzesien Changed to setting: `settings.MAJOR_MINOR_GOLEM_MESSAGES_VERSION` If better name will come to your mind feel free to say it because I don't have any better idea for it and I still don't like this name :P

Reply:
I think at some point we wanted to have pure values in settings only, and keep calculations like this in constants. I am OK with how you have done it currently, because I am used to make calculations and function calls in settings, but please make sure with @cameel .

Reply:
@kbeker
> If better name will come to your mind feel free to say it because I don't have any better idea for it and I still don't like this name :P

In deployment I'm calling it simply _protocol version_ because the first two components of GM version determine whether the protocol is the same or not (see https://github.com/golemfactory/concent-deployment/issues/346).

@kbeker @rwrzesien 
> I think at some point we wanted to have pure values in settings only, and keep calculations like this in constants. I am OK with how you have done it currently, because I am used to make calculations and function calls in settings, but please make sure with @cameel .

Good point @rwrzesien. Thanks for pinging me. You're right, I think these should not be in settings but in this case it's not just about calculations. The fact that just importing `golem_messages.__version__` may read/create files is bad too but even if that was not the case, storing versions in a setting does not make any sense to me:
- Do we even support changing it? Having a setting for it suggests that we do.
    - If not, it should be a constant. Settings are more "public" than constants. They can even be considered a kind of an API and that comes with all the drawbacks of having to keep an API backwards-compatible.
- Will the code even still work when it's changed? Do we have proper checks/validations against values that could break the app?
    - Constants do not need such extensive checks because we have complete control over them. You can even make assertions about them. No so with settings.
    - A setting requires a bigger upfront investment from us to make sure our code works with all supported values. With a constant it's always a single value and you know that if anyone wants to change it, the code must be modified and go through all the usual testing and Q&A to make sure it still works.
- Does changing it even make sense? Will anyone realistically need to set it to anything other than `golem_messages.__version__`?
    - Sometimes it's useful to expose something as a setting even if it's just a temporary or internal thing because it's still useful in our limited circumstances. It's not the case here. I can't think of a situation where we'd want to manually set GM version in our deployment.

It's not even more convenient or readable than simply using `golem_messages.__version__`. In the code. So why do it?

I think these settings should be removed.

Reply:
Issue created https://github.com/golemfactory/concent/issues/1111.